openmdao/utils/shell_proc.py

```diff
@@ -357,3 +359,35 @@ def check_call(args, stdin=None, stdout=None, stderr=None, env=None,
     return_code, error_msg = process.wait(poll_delay, timeout)
     if return_code:
         raise CalledProcessError(return_code, args, error_msg)
+
+
+def _is_process_running(pid):
+    if sys.platform == "win32":
+        # PROCESS_QUERY_LIMITED_INFORMATION is available on Windows Vista and later.
+        PROCESS_QUERY_LIMITED_INFORMATION = 0x1000
+
+        # Attempt to open the process.
+        handle = ctypes.windll.kernel32.OpenProcess(
+            PROCESS_QUERY_LIMITED_INFORMATION, False, pid
+        )
+        if handle:
+            ctypes.windll.kernel32.CloseHandle(handle)
+            return True
+        else:
+            # If OpenProcess fails, check if it's due to access being denied.
+            ERROR_ACCESS_DENIED = 5
+            if ctypes.windll.kernel32.GetLastError() == ERROR_ACCESS_DENIED:
+                return True
+            return False
+    else:
+        try:
+            os.kill(pid, 0)
```

Comment:
Maybe just put a comment here indicating that calling kill with 0 doesn't actually send a kill signal to the process.